dspace-api/src/main/java/org/dspace/identifier/IdentifierServiceImpl.java

```diff
@@ -294,4 +298,61 @@ public void delete(Context context, DSpaceObject dso, String identifier)
             }
         }
     }
+
+    /**
+     * Test whether an identifier provider been disabled in the container of
+     * this object.  If the object is not an Item, providers cannot be disabled.
+     * Otherwise, if *all* containers are marked as disabling generation of this
+     * provider's type of identifier, then generation is disabled.  A container
+     * is marked to disable a provider's identifier type if it has a metadata
+     * field 'local.identifier_generation_disabled.TYPE' where TYPE is the
+     * {@link IdentifierProvider#getIdentifierTypeName() "type name"} of the
+     * provider and the field's first value is "true".
+     *
+     * @return false if this provider is disabled.
+     */
+    private boolean isProviderEnabled(DSpaceObject dso, IdentifierProvider provider)
+            throws SQLException
+    {
+        if (!(dso instanceof Item)) return true;
+
+        Item item = (Item)dso;
+        String providerTypeName = provider.getIdentifierTypeName();
+        log.debug("Is provider {} enabled for Item {}?",
+                providerTypeName, dso.getID());
+
+        List<Collection> collections = item.getCollections();
+        log.debug("Item is in {} collections", collections.size());
+        // Provider is enabled if there are no collections to disable it
+        if (collections.isEmpty())
+        {
+            log.debug("Returning true");
+            return true;
+        }
+
+        boolean disabled = true;
+        for (Collection collection : collections)
+        {
+            log.debug("Checking Collection {} {}",
+                    collection.getID(), collection.getName());
+            CollectionService collectionService = contentServiceFactory.getCollectionService();
+            List<MetadataValue> flagValues = collectionService.getMetadata(collection, "local",
```

Comment:
The metadata type configured below is called: *dspace*.identifier_generation_disabled not *local*.identifier_generation_disabled.